### ISSUE TITLE
Get list of all scss variables with their compiled value

### ISSUE THREAD

**Issue by ecksite** [leafo/scssphp#648]
I wondered if its possible to get a list of all sass variables, defined in the scss files (i.e. as key => value pair in an array) that should be compiled. It seems not to be possible "out of the box", and I tried to create a method by myself but after hours of investigation of the compiler/parsers code, I am more and more frustrated. 

Maybe there is someone, who could give me a little hint how I could do this or has tried something similar?

What I want to achieve:
I’d like to create a plugin that offers the user a GUI for dynamically changing the variables of a CSS Framework (Uikit/Bootstrap) with live preview (some kind of customizer). Therefore i am going to use a scss js compiler client-side.

Serverside compiling via leafos scssphp would only be used for getting all variable definitions and their component-wise default values (so the value should be compiled). 
Then I would use those information for setting up the customizer settings fields (like color options or font size selects) for each separated  component (i.e. Global colors, Navbar, Sections, Buttons). This way the user could set up different css frameworks to start with, or even create one himself and the customizer field settings were quite easy to maintain.

Getting all variables could be achieved via some regex and is not the problem, but in order to get the initial value for those fields requires compiling.

**Comment by Cerdic** [comment]
I guess a way would be to add a comment section to your compiled file in order to display each variable value, but at the moment interpolation of vars in comment is not working (see #565)

**Comment by robocoder** [comment]
Ok, so you understand the catch-22. You can't get the compiled value without compiling, and values can change during the compilation.

You could try subclassing the Compiler class. Compile the scss files but throwaway the output. There you can inspect the variables stored at `$this->rootEnv`.

**Comment by ecksite** [comment]
It's totally fine that values can change during the compilation.
I just want to compile once, so I will get the reduced and final css property value (i.e. 5px, rgba(244,100,0,0.2), 1px solid red, #00eeff). 
Of course this means that there could be values, that inherit from each other, so if i had a variable named `@global-color-blue` with value`#0000ff` and another variable named `@component-color-primary`with value of `@global-color-blue` both would have the value `#0000ff`.
And of course there could be values, that get overridden by other value assignments in another scss file, so if i had a variable named `@global-color-blue` with initial value`#0000ff` and later there was a file which assignes the `@global-color-blue` a value of `#ffffff` the variables value after compiling would be `#ffffff`.

I have already tried creating a subcass, in order to get this working, but to be honest I have given up because a compiler is very complex and its always hard to follow each step in the big picture. 
I thought, maybe one of the contributers, that might have a deeper understanding of the implementation could give me hint or could implement it without much effort because of his knowledge of the compilers implementation.

I guess the problem is something that a few people would like to see solved (currently 3 likes (?) :) ).
